taskcat/testing/manager.py

```diff
@@ -0,0 +1,252 @@
+                self.test_definition.delete_stacks({"status": "CREATE_COMPLETE"})
+        else:
+            self.test_definition.delete_stacks()
+
+        if not dont_wait_for_delete:
+            terminal_printer.report_test_progress(stacker=self.test_definition)
+
+        # TODO: summarise stack statusses (did they complete/delete ok) and print any
+        #  error events
+
+        # Delete Templates and Buckets
+        buckets = self.config.get_buckets()
+
+        if not no_delete or (keep_failed is True and len(status["FAILED"]) == 0):
+            deleted: ListType[str] = []
+            for test in buckets.values():
+                for bucket in test.values():
+                    if (bucket.name not in deleted) and not bucket.regional_buckets:
+                        bucket.delete(delete_objects=True)
+                        deleted.append(bucket.name)
+
+        # 9. raise if something failed
+        # - grabbing the status again to ensure everything deleted OK.
+
+        status = self.test_definition.status()
+        if len(status["FAILED"]) > 0:
+            raise TaskCatException(
+                f'One or more stacks failed tests: {status["FAILED"]}'
+            )
+
+    def report(
+        self, output_directory: str = "./taskcat_outputs",
+    ):
+        """
+        Generates a report of the status of Cloudformation stacks.
+        """
+        report_path = Path(output_directory).resolve()
+        report_path.mkdir(exist_ok=True)
+        cfn_logs = _CfnLogTools()
+        cfn_logs.createcfnlogs(self.test_definition, report_path)
+        ReportBuilder(
+            self.test_definition, report_path / "index.html"
+        ).generate_report()
+
+
+def _trim_regions(regions, config):
+    if regions != "ALL":
+        for test in config.config.tests.values():
+            to_pop = []
+            idx = 0
+            if test.regions:
+                for _ in test.regions:
+                    if test.regions[idx] not in regions.split(","):
+                        to_pop.append(idx)
+                    idx += 1
+                to_pop.reverse()
+                for idx in to_pop:
+                    test.regions.pop(idx)
+
+
+def _trim_tests(test_names, config):
+    if test_names != "ALL":
+        for test in list(config.config.tests):
+            if test not in test_names.split(","):
+                del config.config.tests[test]
+
+
+def _build_args(enable_sig_v2, regions, default_profile):
+    args: Dict[str, Any] = {}
+    if enable_sig_v2:
+        args["project"] = {"s3_enable_sig_v2": enable_sig_v2}
+    if regions != "ALL":
+        if "project" not in args:
+            args["project"] = {}
+        args["project"]["regions"] = regions.split(",")
+    if default_profile:
+        _auth_dict = {"default": default_profile}
+        if not args.get("project"):
+            args["project"] = {"auth": _auth_dict}
+        else:
+            args["project"]["auth"] = _auth_dict
+    return args
```

Comment:
This one also seems like it could be handled elsewhere. Its just defaulting certain parts of the config dictionary?? Not sure really.

Reply:
Context:

One of the big things we wanted to avoid in `0.9.0+` is functional drift between CLI arguments and the config file. We struggled with that previously. Our current model forces all CLI arguments to have an underlying config file option. To pair the two together, any CLI arguments are transformed into a new config dict, and we re-create the `Config` object with those new options. 

This is specific to the `Test` CLI command, but there is most-likely a better way to handle manipulating the config to reflect CLI arguments. My first thought is a decorator or something similar that hides most of this heavy lifting. Longer term (v1.0.0+ release), I'd like to make this low-effort to interface with. 

Anyway, that's why that's there. Same with `_trim_*`.